packages/target-tauri/src-tauri/src/html_window/commands.rs

```diff
@@ -0,0 +1,167 @@
+        HTML_EMAIL_ALWAYS_ALLOW_REMOTE_CONTENT_KEY, HTML_EMAIL_WARNING_DEFAULT,
+        HTML_EMAIL_WARNING_KEY,
+    },
+    HtmlEmailInstancesState,
+};
+
+use super::error::Error;
+
+async fn set_load_remote_content(webview: &tauri::Webview, new_state: bool) -> Result<(), Error> {
+    let instance_state = webview.state::<HtmlEmailInstancesState>();
+    let instance = instance_state
+        .get(webview.window().label())
+        .await
+        .ok_or(Error::WindowNotFoundInState)?;
+
+    if instance.blocked_by_proxy {
+        return Err(Error::BlockedByProxy);
+    }
+
+    instance_state
+        .set_network_allow_state(webview.window().label(), new_state)
+        .await;
+
+    // reload header & html email webviews to apply changes
+    for webview in webview.window().webviews() {
+        if webview.label().ends_with("-mail") {
+            webview.reload()?;
+        }
+    }
+    Ok(())
+}
+
+#[tauri::command]
+pub(crate) async fn html_email_set_load_remote_content(
+    app: tauri::AppHandle,
+    webview: tauri::Webview,
+    load_remote_content: bool,
+) -> Result<(), Error> {
+    let desktop_settings = app.store(CONFIG_FILE)?;
+    let warning = get_setting_bool_or(
+        desktop_settings.get(HTML_EMAIL_WARNING_KEY),
+        HTML_EMAIL_WARNING_DEFAULT,
+    );
+
+    if warning && load_remote_content {
+        let (tx, rx) = tokio::sync::oneshot::channel::<bool>();
+
+        app.dialog()
+            .message("tx('load_remote_content_ask')")
+            .parent(&webview.window())
+            .buttons(tauri_plugin_dialog::MessageDialogButtons::OkCancelCustom(
+                // TODO use translation strings, as soon as translations are avaialble in rust backend
+                "Yes".to_owned(),
+                "No".to_owned(),
+            ))
+            .show(|answer| tx.send(answer).unwrap());
+        if !rx.await? {
+            return Err(Error::UserCanceled);
+        }
+    }
+
+    set_load_remote_content(&webview, load_remote_content).await
+}
+
+#[tauri::command]
+pub(crate) fn html_email_open_menu(
+    app: tauri::AppHandle,
+    webview: tauri::Webview,
+    html_instances_state: tauri::State<HtmlEmailInstancesState>,
+) -> Result<(), Error> {
+    let desktop_settings = app.store(CONFIG_FILE)?;
+    let always_load = get_setting_bool_or(
+        desktop_settings.get(HTML_EMAIL_ALWAYS_ALLOW_REMOTE_CONTENT_KEY),
+        HTML_EMAIL_ALWAYS_ALLOW_REMOTE_CONTENT_DEFAULT,
+    );
+    let warning = get_setting_bool_or(
+        desktop_settings.get(HTML_EMAIL_WARNING_KEY),
+        HTML_EMAIL_WARNING_DEFAULT,
+    );
+
+    let instance_state = block_on(html_instances_state.get(webview.window().label()))
+        .ok_or(Error::WindowNotFoundInState)?;
+
+    let always_load_remote_images = CheckMenuItem::new(
+        &app,
+        "tx('always_load_remote_images')",
+        true,
+        always_load,
+        None::<&str>,
```

Comment:
Quick rust question. What's this? Is that passing in a `None` but typed as a `&str`?

Reply:
a turbofish https://turbo.fish/about
basically this syntax sets the type of the option. it is also sometimes used when collecting an iterator https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect